includes/api/class-wc-admin-rest-reports-export-controller.php

```diff
@@ -0,0 +1,248 @@
+	}
+
+	/**
+	 * Get the Export status schema, conforming to JSON Schema.
+	 *
+	 * @return array
+	 */
+	public function get_export_status_public_schema() {
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/draft-04/schema#',
+			'title'      => 'report_export_status',
+			'type'       => 'object',
+			'properties' => array(
+				'percent_complete' => array(
+					'description' => __( 'Percentage complete.', 'woocommerce-admin' ),
+					'type'        => 'int',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'download_url'     => array(
+					'description' => __( 'Export download URL.', 'woocommerce-admin' ),
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+					'readonly'    => true,
+				),
+			),
+		);
+
+		return $this->add_additional_fields_schema( $schema );
+	}
+
+	/**
+	 * Export data based on user request params.
+	 *
+	 * @param  WP_REST_Request $request Request data.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function export_items( $request ) {
+		$report_type = $request['type'];
+		$report_args = empty( $request['report_args'] ) ? array() : $request['report_args'];
+		$export_id   = str_replace( '.', '', microtime( true ) );
+
+		$total_rows = WC_Admin_Report_Exporter::queue_report_export( $export_id, $report_type, $report_args );
+
+		if ( 0 === $total_rows ) {
+			$response = rest_ensure_response(
+				array(
+					'status'  => 'error',
+					'message' => __( 'There is no data to export for the given request.', 'woocommerce-admin' ),
+				)
+			);
+
+		} else {
+			$response = rest_ensure_response(
+				array(
+					'status'  => 'success',
+					'message' => __( 'Your report file is being generated.', 'woocommerce-admin' ),
+				)
+			);
+
+			// Include a link to the export status endpoint.
+			$response->add_links(
+				array(
+					'status' => array(
+						'href' => rest_url( sprintf( '%s/reports/%s/export/%s/status', $this->namespace, $report_type, $export_id ) ),
+					),
+				)
+			);
+
+			WC_Admin_Report_Exporter::update_export_percentage_complete( $report_type, $export_id, 0 );
+		}
+
+		$data = $this->prepare_response_for_collection( $response );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Export status based on user request params.
+	 *
+	 * @param  WP_REST_Request $request Request data.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function export_status( $request ) {
+		$report_type = $request['type'];
+		$export_id   = $request['export_id'];
+		$percentage  = WC_Admin_Report_Exporter::get_export_percentage_complete( $report_type, $export_id );
+
+		if ( false === $percentage ) {
+			return new WP_Error(
+				'woocommerce_admin_reports_export_invalid_id',
+				__( 'Sorry, there is no export with that ID.', 'woocommerce-admin' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		$result = array(
+			'percent_complete' => $percentage,
+		);
+
+		// @todo - add thing in the links below instead?
```

Comment:
add thing?

Reply:
Ha! Not my best comment..

I meant here: "should we add the download URL in the `_links` instead?